warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -896,8 +896,8 @@ else if (key.compareTo(this.lastRangeSeeked.getStartKey()) < 0) {
                     throw new RuntimeException("Ivarator query was cancelled");
                 }
 
-                // if we have any persisted data or we have scanned a significant number of keys, then persist it completely
-                if (this.set != null && (this.set.hasPersistedData() || (this.scannedKeys.get() >= this.scanThreshold))) {
+                // persist set to disk so we can reuse it on a rebuild or yield
+                if (this.set != null) {
                     forcePersistence();
```

Comment:
Is it possible for the ivarator to have found no results leaving the set null?  If that is the case then we need to make sure the "complete" flag is generated regardless whether we have a set.  I think it is correct but I need to make sure..... investigating.

Reply:
Ah, I remember now.  The set will be non-null in the normal case.  Its the "unsorted" case where it is null so this is correct.